### PR TITLE
[0.3] Backport Go modules (#888)

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@vmware.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

File: go.vendor [hunk 1]
```diff
@@ -0,0 +1 @@
+sigs.k8s.io/cluster-api
```
Comment (ncdc) [inline go.vendor]: What is this for?

Comment (vincepri) [reply go.vendor]: https://github.com/kubernetes-sigs/cluster-api/pull/1134

**Comment by ncdc** [comment]
/lgtm
/hold

@detiber are you interested in reviewing?

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+	k8s.io/klog v0.3.2
+	k8s.io/kubernetes v1.13.3
+	sigs.k8s.io/cluster-api v0.1.8
+	sigs.k8s.io/controller-runtime v0.1.12
+	sigs.k8s.io/controller-tools v0.1.11
+	sigs.k8s.io/testing_frameworks v0.1.1
+	sigs.k8s.io/yaml v1.1.0
+)
+
+replace k8s.io/apimachinery => k8s.io/apimachinery v0.0.0-20190221213512-86fb29eff628
```
Comment (detiber) [inline go.mod]: Do we need to list `cluster-api` as a replace to avoid issues after we start cutting v1alpha2-related releases?

Comment (vincepri) [reply go.mod]: I'd think that given the incompatibilities it'll not default to v2. We can add a replace at some point if needed though

Comment (detiber) [reply go.mod]: I'm not sure how much compatibility testing is done with gomod, especially prior to major versions being set. It definitely will not attempt to compile against versions like `dep` does.

Comment (vincepri) [reply go.mod]: We can follow-up in another commit if needed, I opened this mostly to backport the modules support before the release is cut.

**Pull request merged**